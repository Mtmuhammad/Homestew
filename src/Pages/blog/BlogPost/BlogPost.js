import React from "react";
import "./BlogPost.scss";

const BlogPost = ({title, author, img, comments, likes, date}) => {
  return (
    
    <div className="col-12 col-md-6 col-lg-4">
      <div className="blog-post">
        <div className="blog-post-thumb">
          <img
            className="img-fluid"
            src={img}
            alt={title}
          />
        </div>

        <div className="mt-4">
          <div className="justify-content-between align-items-center d-flex">
            <div className="d-flex">
              <div className="blog-post-author">
                <a href="#0">By {author}</a>
              </div>

              <div className="blog-post-date">
                <a href="#0">{date}</a>
              </div>
            </div>

            <div className="d-flex">
              <div className="blog-post-fav">
                <a href="#0">
                  <i className="fa fa-heart-o" aria-hidden="true"></i> {likes}
                </a>
              </div>

              <div className="blog-post-comments">
                <a href="#0">
                  <i className="fa fa-comment-o" aria-hidden="true"></i> {comments}
                </a>
              </div>

              <div className="blog-post-share">
                <a href="#0">
                  <i className="fa fa-share-alt" aria-hidden="true"></i>
                </a>
              </div>
            </div>
          </div>
          <a className="blog-post-headline" href="#0">
            <h4 >
              {title}
            </h4>
          </a>
        </div>
      </div>
    </div>
    
  );
};

export default BlogPost;
