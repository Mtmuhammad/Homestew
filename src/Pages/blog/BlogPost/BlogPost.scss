@use "../../../Styles/abstracts/mixins" as *;
@use "../../../Styles/abstracts/variables" as *;

.blog-post {
  margin-bottom: 6rem;

  & img {
    border-radius: 5px;
  }

  & h4 {
    margin: 10px 0;
    font-size: 3rem;
    transition: all 0.3s;
  }
  &-author {
    & a {
      margin-right: 30px;
      position: relative;
      z-index: 1;
      font-size: 1.4rem;
      color: $blog-grey;

      &:hover {
        color: $primary-font;
      }

      &:after {
        width: 2px;
        height: 60%;
        content: "";
        position: absolute;
        top: 20%;
        right: -14px;
        z-index: 2;
        background-color: $blog-grey;
      }
    }
  }

  &-date a,
  .blog-post-share a {
    color: $blog-grey;
    font-size: 1.4rem;
  }

  &-fav a,
  .blog-post-comments a {
    color: $blog-grey;
    margin-right: 15px;
    font-size: 1.4rem;
  }

  &-headline {
    color: $primary-font;
  }
}
