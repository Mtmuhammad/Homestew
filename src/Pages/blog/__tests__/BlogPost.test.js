import React from "react";
import { render, screen } from "@testing-library/react";
import BlogPost from "../BlogPost/BlogPost";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(
      <BlogPost
            title={"How to Properly Cut and Prep Your Vegetables"}
            author={"Mike Jones"}
            likes={14}
            comments={26}
            date={"Jan 11, 2022"}
            img={"Blog1.jpg"}
          />
    );
  });

  test("should match snapshot", () => {
    const { asFragment } = render(
      <BlogPost
            title={"How to Properly Cut and Prep Your Vegetables"}
            author={"Mike Jones"}
            likes={14}
            comments={26}
            date={"Jan 11, 2022"}
            img={"Blog1.jpg"}
          />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});



test("should render img", () => {
   render(
      <BlogPost
      title={"How to Properly Cut and Prep Your Vegetables"}
      author={"Mike Jones"}
      likes={14}
      comments={26}
      date={"Jan 11, 2022"}
      img={"Blog1.jpg"}
    />
   );
 
   const img = screen.getByAltText(/How to Properly Cut and Prep Your Vegetables/i);
   expect(img).toBeInTheDocument();
   expect(img).toHaveClass("img-fluid");
 });
 
 
 test("should render heading", () => {
   render(
      <BlogPost
      title={"How to Properly Cut and Prep Your Vegetables"}
      author={"Mike Jones"}
      likes={14}
      comments={26}
      date={"Jan 11, 2022"}
      img={"Blog1.jpg"}
    />
   );
 
   const category = screen.getByRole("heading");
   expect(category).toBeInTheDocument();
   expect(category).toHaveTextContent(/How to Properly Cut and Prep Your Vegetables/i);
 });
 