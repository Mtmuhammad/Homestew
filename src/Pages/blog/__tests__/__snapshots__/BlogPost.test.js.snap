// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-12 col-md-6 col-lg-4"
  >
    <div
      class="blog-post"
    >
      <div
        class="blog-post-thumb"
      >
        <img
          alt="How to Properly Cut and Prep Your Vegetables"
          class="img-fluid"
          src="Blog1.jpg"
        />
      </div>
      <div
        class="mt-4"
      >
        <div
          class="justify-content-between align-items-center d-flex"
        >
          <div
            class="d-flex"
          >
            <div
              class="blog-post-author"
            >
              <a
                href="#0"
              >
                By Mike Jones
              </a>
            </div>
            <div
              class="blog-post-date"
            >
              <a
                href="#0"
              >
                Jan 11, 2022
              </a>
            </div>
          </div>
          <div
            class="d-flex"
          >
            <div
              class="blog-post-fav"
            >
              <a
                href="#0"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-heart-o"
                />
                 14
              </a>
            </div>
            <div
              class="blog-post-comments"
            >
              <a
                href="#0"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-comment-o"
                />
                 26
              </a>
            </div>
            <div
              class="blog-post-share"
            >
              <a
                href="#0"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-share-alt"
                />
              </a>
            </div>
          </div>
        </div>
        <a
          class="blog-post-headline"
          href="#0"
        >
          <h4>
            How to Properly Cut and Prep Your Vegetables
          </h4>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
